### PR TITLE
First version of node exporter

### PR DESCRIPTION
**Pull request by Rados13**

### CODE REVIEW

File: grafana/provisioning/dashboards/metrics.json [hunk 1]
```diff
@@ -907,56 +915,571 @@
...
+                "description": "Allows you limit graphs only to selected nodes",
+                "hide": 0,
+                "includeAll": true,
+                "label": "Node",
+                "multi": true,
+                "name": "node",
+                "options": [],
+                "query": {
+                    "qryType": 1,
+                    "query": "label_values(node_uname_info,instance)",
```
Comment (Davenury) [inline grafana/provisioning/dashboards/metrics.json:1482]: I'm not sure if instance regards actual node or instance of node exporter (IMO the latter). Better pick (IMHO) would be `nodename` or `domainname` (node name is the value of `hostname` typed in the container, whereas I don't have domainname).

File: grafana/provisioning/dashboards/metrics.json [hunk 2]
```diff
@@ -907,56 +915,571 @@
...
+                    "value": [
+                        "$__all"
+                    ]
+                },
+                "datasource": {
+                    "type": "prometheus",
+                    "uid": "PBFA97CFB590B2093"
+                },
+                "definition": "label_values(node_uname_info,instance)",
+                "description": "Allows you limit graphs only to selected nodes",
```
Comment (Davenury) [inline grafana/provisioning/dashboards/metrics.json:1473]: `... in node-related row`?

File: grafana/provisioning/dashboards/metrics.json [hunk 3]
```diff
@@ -907,56 +915,571 @@
...
+                        }
+                    },
+                    "targets": [
+                        {
+                            "datasource": {
+                                "type": "prometheus",
+                                "uid": "${ds}"
+                            },
+                            "editorMode": "code",
+                            "expr": "100 - (avg by (instance) (irate(node_cpu_seconds_total{mode=\"idle\", instance=~\"$node\"}[1m])) * 100)",
```
Comment (Davenury) [inline grafana/provisioning/dashboards/metrics.json:1016]: How did you get this query? Why mode `idle`? I would also maybe advise to use `$__rate_interval` instead of `1m` - I'm not sure how it would behave if e.g. someone had their scrapping interval set to 5 minutes.

File: grafana/provisioning/dashboards/metrics.json [hunk 4]
```diff
@@ -907,56 +915,571 @@
...
+                    ],
+                    "title": "Average bytes transmitted by node ($node)",
+                    "type": "timeseries"
+                },
+                {
+                    "datasource": {
+                        "type": "prometheus",
+                        "uid": "${ds}"
+                    },
+                    "description": "Memory usage from each of the nodes.",
```
Comment (Davenury) [inline grafana/provisioning/dashboards/metrics.json:1217]: There's also a possibility to display a threshold of maximum memory of the node out of a query. It's a rather advanced topic, so if you'd like that, tell me and I'll show you.

File: grafana/provisioning/dashboards/metrics.json [hunk 5]
```diff
@@ -907,56 +915,571 @@
...
+                        }
+                    },
+                    "targets": [
+                        {
+                            "datasource": {
+                                "type": "prometheus",
+                                "uid": "${ds}"
+                            },
+                            "editorMode": "code",
+                            "expr": "(avg by (instance) (node_memory_MemTotal_bytes{instance=~\"$node\"} - node_memory_MemFree_bytes{instance=~\"$node\"} - node_memory_Buffers_bytes{instance=~\"$node\"} - node_memory_Cached_bytes{instance=~\"$node\"}))",
```
Comment (Davenury) [inline grafana/provisioning/dashboards/metrics.json:1295]: Seems overly complicated - isn't there a better way to do that? (e.g. node_memory_allocated or something - idk tbh)

File: grafana/provisioning/dashboards/metrics.json [hunk 6]
```diff
@@ -907,56 +915,571 @@
...
+                            "legendFormat": "{{instance}}",
+                            "range": true,
+                            "refId": "A"
+                        }
+                    ],
+                    "title": "Number of allocated file descriptors by node ($node)",
+                    "type": "timeseries"
+                }
+            ],
+            "title": "Node stats",
```
Comment (Davenury) [inline grafana/provisioning/dashboards/metrics.json:1399]: You could think about the presentation of those graphs. Rn they are one under another and it's giving repeatable and a bit boring (IMHO).

File: grafana/provisioning/dashboards/metrics.json [hunk 7]
```diff
@@ -23,183 +23,187 @@
...
-                        "showPoints": "never",
-                        "spanNulls": false,
-                        "stacking": {
-                            "group": "A",
-                            "mode": "none"
+                    "fieldConfig": {
+                        "defaults": {
+                            "color": {
+                                "fixedColor": "dark-blue",
+                                "mode": "shades"
```
Comment (Davenury) [inline grafana/provisioning/dashboards/metrics.json:44]: idk from where you got that, but I got an error:
Error: "shades" not found in: fixed,thresholds,palette-classic,continuous-GrYlRd,continuous-RdYlGr,continuous-BlYlRd,continuous-YlRd,continuous-BlPu,continuous-YlBl,continuous-blues,continuous-reds,continuous-greens,continuous-purples

trying to edit the panel. Deleting the `color` object from json view helped but I think it's something that should be addressed.

Also, for this panel you can use `Stat` visualization, not `Gauge` - I think it would go better.

**Review by Davenury — COMMENTED** [review]
Some of my thoughts about the dashboard. Key notes:
* dashboard seems a bit repetitive - it's not necessarily a bad thing, I just wanted to point it out - think if all of the visualizations must be one under another. Some might be fine being next to each other, etc.
* I do believe that queries are created correctly - I didn't check them
* I'd advise to change `1m` to `$__rate_interval` - it's an variable created automatically by Grafana and might come in handy when scrape interval is larger than 1m (dunno though, just my guess on how does it work)

File: grafana/provisioning/dashboards/metrics.json (hunk 1 above)
Comment (Rados13) [reply grafana/provisioning/dashboards/metrics.json:1482]: I chose `instance` because it is easier to understand on which node this is happening and compare it with stats from our containers.

File: grafana/provisioning/dashboards/metrics.json (hunk 3 above)
Comment (Rados13) [reply grafana/provisioning/dashboards/metrics.json:1016]: I got it from ChatGPT. The mode is set to idle because we calculate how much time the CPU spends idle. We assume that all other time is used for working, so to calculate what percentage of time CPU is used we subtract the percentage of time it spends idle from 100.

File: grafana/provisioning/dashboards/metrics.json (hunk 4 above)
Comment (Rados13) [reply grafana/provisioning/dashboards/metrics.json:1217]: I think that it is a great future improvement.

File: grafana/provisioning/dashboards/metrics.json (hunk 5 above)
Comment (Rados13) [reply grafana/provisioning/dashboards/metrics.json:1295]: I didn't find anything like this most proposed queries look similar to that e.g:
- https://gist.github.com/payam-int/edf977c6af603fee0ce1b05da7792fe7
- https://stackoverflow.com/a/55451854
(The updated version from stack is very similar to gist)

File: grafana/provisioning/dashboards/metrics.json (hunk 6 above)
Comment (Rados13) [reply grafana/provisioning/dashboards/metrics.json:1399]: Yeah, it is boring, but I don't know if Grafana dashboards must be entertaining. When I need entertainment, I read books, play games, and definitely not read Grafana dashboards 😆 .

File: docker-compose.yaml [hunk 8]
```diff
@@ -83,7 +83,8 @@ services:
       context: ./prometheus
       dockerfile: Dockerfile
       args:
-        TARGETS: ${PROMETHEUS_TARGETS:-jellyfish1:9568,jellyfish2:9568}
+        JELLYFISH_TARGETS: ${PROMETHEUS_TARGETS:-jellyfish1:9568,jellyfish2:9568}
+        NODE_TARGETS: '172.17.0.1:9100'
```
[the remark this answers is not in the document]
Comment (Rados13) [reply docker-compose.yaml:87]: Node-exporter, to get access to more statistics regarding the network, must have access to the OS network; that's why it is run in `network_mode: host`. When it is run in this mode, it can't be referred by other services via `service_name,` as other containers don't run in `network_mode: host`. Furthermore, They don't have access through localhost to services run on the host OS. To allow this, docker added a special IP address that allows access services run on the host OS and this IP address is `172.17.0.1`.
So to summarize `172.17.0.1` in docker containers allows access services run on `127.0.0.1` in the host OS.

File: .github/workflows/production_build_and_deploy.yml [hunk 9]
```diff
@@ -32,7 +32,8 @@ jobs:
           BE_HOST=${{env.BE_HOST}}
           BE_JF_SECURE_CONNECTION=true
           BE_JF_ADDRESSES=${{env.JF1_HOST}} ${{env.JF2_HOST}}
-          PROMETHEUS_TARGETS=jellyfish1:9568,${{secrets.PRODUCTION2_IP}}:9568
+          JELLYFISH_TARGETS=jellyfish1:9568,${{secrets.SANDBOX2_IP}}:9568
+          NODE_TARGETS=172.17.0.1:9100,${{secrets.SANDBOX2_IP}}:9100
```
[the remark this answers is not in the document]
Comment (Rados13) [reply .github/workflows/production_build_and_deploy.yml:36]: I described in a comment below why we use this IP address.

File: docker-compose.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (mickel8) [reply docker-compose.yaml]: Could we please add a short comment?

**Review by mickel8 — APPROVED** [review]

**Review by blazpie — APPROVED** [review]

File: grafana/provisioning/dashboards/metrics.json (hunk 1 above)
Comment (Davenury) [reply grafana/provisioning/dashboards/metrics.json:1479]: I proposed `nodename` since that's what descriptive name of the node is in my company's metrics. My only fear here was if `instance` wasn't pointing to node exporter. Although node exporters probably work one by node and those should be easy to find like that.

File: grafana/provisioning/dashboards/metrics.json (hunk 6 above)
Comment (Davenury) [reply grafana/provisioning/dashboards/metrics.json:1396]: Lol, you do what? Don't tell me you don't look at Grafana dashboards when bored :scream:

**Review by Davenury — APPROVED** [review]
